subsys/ipc/ipc_service/backends/ipc_rpmsg_static_vrings.c

```diff
@@ -448,10 +469,86 @@ static int open(const struct device *instance)
 
 }
 
+static int get_tx_buffer(const struct device *instance, void *token,
+			 void **data, uint32_t *len, k_timeout_t wait)
+{
+	struct ipc_rpmsg_ept *rpmsg_ept;
+	uint32_t max_size;
+	void *payload;
+	int bin_wait;
+
+	if (!data || !len) {
+		return -EINVAL;
+	}
+
+	/* OpenAMP only supports a binary wait / no-wait */
+	if (!K_TIMEOUT_EQ(wait, K_FOREVER) && !K_TIMEOUT_EQ(wait, K_NO_WAIT)) {
+		return -EINVAL;
+	}
+
+	bin_wait = K_TIMEOUT_EQ(wait, K_FOREVER);
+	rpmsg_ept = (struct ipc_rpmsg_ept *) token;
+
+	payload = rpmsg_get_tx_payload_buffer(&rpmsg_ept->ep, &max_size, bin_wait);
+	if (!payload) {
+		return -EIO;
+	}
+
+	/*
+	 * We still want to return the payload buffer in case of -ENOMEM
+	 * because the user would want to drop it or just use it anyway with
+	 * smaller data.
+	 */
+	(*data) = payload;
+
+	if (*len > max_size) {
+		*len = max_size;
```

Comment:
isn't this a problem? We allocated too small buffer and there is no option to return it  (dropping tx buffer is not supported).
Is there a way to get `max_size` prior to calling `rpmsg_get_tx_payload_buffer` and return error if it is less than requested value?

Reply:
no. But this will be not an issue when OpenAMP will support the buffer drop (see https://github.com/OpenAMP/open-amp/issues/361)